### ISSUE TITLE
Advanced playground should use JSON instead of XML on reload

### ISSUE THREAD

**Issue by BeksOmega** [rachel-fenichel/blockly-samples#1049]
**Category**

 - Plugins

**Component**

dev-tools, in the advanced playground.

**Describe the bug**

~The playground [still uses XML serialization](https://github.com/google/blockly-samples/blob/03c119bc5579393860b717f8ae6f265d6840f955/plugins/dev-tools/src/playground/options.js#L135) on reload, so if you test out readOnly mode, your blocks get stuck as readonly. The JSON serializer doesn't have this problem, so we should switch to using that.~

The playground still uses XML serialization during initailization. This means that custom serializers are not deserialized (e.g. the suggested-blocks plugin, or the backpack). We should use JSON serialization by default in the advanced playground.

**To Reproduce**

1. CD into `plugins/dev-tools`.
2. Run `npm run start`.
3. Add the following JSON to the JSON tab:
```json
{
  "blocks": {
    "languageVersion": 0,
    "blocks": [
      {
        "type": "controls_if",
        "x": 0,
        "y": 0
      }
    ]
  }
}
```
4. Reload the page.
5. Observe how no block exists in the workspace, when one should.

## To Fix

1) Run through the reproduction steps to make sure you can reproduce the problem.
1) Change the [playgroundState's default state](https://github.com/google/blockly-samples/blob/11e2600f4b8391b82a67cb14dcf4f1c5e4c0466d/plugins/dev-tools/src/playground/index.js#L124) to have an `activeTab: 'JSON'`, and `workspaceJson: {}` instead of `workspaceXML: ''`.
2) Change the [initialization code](https://github.com/google/blockly-samples/blob/11e2600f4b8391b82a67cb14dcf4f1c5e4c0466d/plugins/dev-tools/src/playground/index.js#L314) to use Json instead of Xml.
3) Change the [updateEditor](https://github.com/google/blockly-samples/blob/11e2600f4b8391b82a67cb14dcf4f1c5e4c0466d/plugins/dev-tools/src/playground/index.js#L218) code to use Json. Change it to use `Blockly.serialization.workspaces.save/load` instead of `workspaceToDom` and `domToWorkspace`.
4) Run through the reproduction steps again, and make sure the error does not reproduce.

**Comment by divyajyotiuk** [comment]
I would like to work on this!

**Comment by BeksOmega** [comment]
@divyajyotiuk Woot go for it!

**Comment by PrathiPuthran** [comment]
@BeksOmega i would like to pair program on this with Divya, I have pinged in slack, thank you

**Comment by AishwaryaBalyaya01** [comment]
I would also like to pair program with you, have pinged you on Slack!

**Comment by BeksOmega** [comment]
There's inconsistency between whether the `workspaceJson` is a string or an object, which is causing errors when you try to load the playground.

**Comment by BeksOmega** [comment]
Closed by #1996
